Plugin/OrderPlugin.php

```diff
@@ -263,4 +281,46 @@ public function afterSetState($subject, $result)
         }
         return $result;
     }
+
+    /**
+     * Replace remote IP with data from bolt additional request params
+     *
+     * @param Order $subject
+     * @param $remoteIp
+     * @return array
+     */
+    public function beforeSetRemoteIp(Order $subject, $remoteIp)
+    {
+        try {
+            $boltAdditionalParams = $this->boltAdditionalRequestParamsReader->getBoltAdditionalParams();
+            if (isset($boltAdditionalParams[self::BOLT_ADDITIONAL_PARAM_CLIENT_IP_KEY])) {
+                $remoteIp = $boltAdditionalParams[self::BOLT_ADDITIONAL_PARAM_CLIENT_IP_KEY];
+            }
+        } catch (\Exception $e) {
+            $this->bugsnag->notifyException($e);
+        }
+
+        return [$remoteIp];
+    }
+
+    /**
+     * Replace x-forwarded-for IP with data from bolt additional request params
+     *
+     * @param Order $subject
+     * @param $xForwardedFor
+     * @return array
+     */
+    public function beforeSetXForwardedFor(Order $subject, $xForwardedFor)
+    {
+        try {
+            $boltAdditionalParams = $this->boltAdditionalRequestParamsReader->getBoltAdditionalParams();
+            if (isset($boltAdditionalParams[self::BOLT_ADDITIONAL_PARAM_CLIENT_IP_KEY])) {
+                $xForwardedFor = $boltAdditionalParams[self::BOLT_ADDITIONAL_PARAM_CLIENT_IP_KEY];
```

Comment:
Unless I'm misunderstanding the situation I believe this is a misuse of the header. The forwarded header is meant to be used for proxy situations to determine who is the original sender. This is a request coming from the backend, which says is coming from the frontend client.